scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,161 @@
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
+
+      if (scriptFailedFlag) {
+        println("REGEX PATTERN CHECKS FAILED")
+        exitProcess(1)
+      } else {
+        println("REGEX PATTERN CHECKS PASSED")
+      }
+    }
+
+    fun checkProhibitedFileNamePattern(
+      repoPath: String,
+      searchFiles: Sequence<File>,
+      prohibitedFilenameRegexString: String,
+      errorToShow: String
+    ): Boolean {
+      var filenamePatternCheckFailedFlag = false
+      val prohibitedFilenameRegex = prohibitedFilenameRegexString.toRegex()
+      searchFiles.forEach {
+        val filePath = it.toString().removePrefix(repoPath)
+        if (prohibitedFilenameRegex.matches(filePath)) {
+          logProhibitedFilenameFailure(
+            filePath = filePath,
+            errorToShow = errorToShow
+          )
+          filenamePatternCheckFailedFlag = true
+        }
+      }
+      return filenamePatternCheckFailedFlag
+    }
+
+
+    fun checkProhibitedContent(
+      repoPath: String,
+      searchFiles: Sequence<File>,
+      fileNameRegexString: String,
+      prohibitedContentRegexString: String,
+      errorToShow: String
+    ): Boolean {
+      var contentCheckFailedFlag = false
+      val fileNameRegex = fileNameRegexString.toRegex()
+      val prohibitedContentRegex = prohibitedContentRegexString.toRegex()
+      searchFiles.forEach {
+        if (fileNameRegex.matches(it.name)) {
+          var lineNumber = 0
+          File(it.toString()).forEachLine { l ->
+            lineNumber++
+            if (prohibitedContentRegex.matches(l)) {
+              logProhibitedContentFailure(
+                lineNumber = lineNumber,
+                errorToShow = errorToShow,
+                filePath = it.toString().removePrefix(repoPath)
+              )
+              contentCheckFailedFlag = true
+            }
+          }
+        }
+      }
+      return contentCheckFailedFlag
+    }
+
+    fun collectSearchFiles(
+      repoPath: String,
+      allowedDirectories: Array<String>,
+      exemptionList: Array<String> = arrayOf<String>()
+    ): Sequence<File> {
+      val validPaths = File(repoPath).walk().filter { it ->
+        checkIfAllowedDirectory(
+          it.toString().removePrefix(repoPath),
+          allowedDirectories)
+          && it.isFile
+          && it.name !in exemptionList
+      }
+      return validPaths
+    }
+
+    fun checkIfAllowedDirectory(
+      pathString: String,
+      allowedDirectories: Array<String>
+    ): Boolean {
+      allowedDirectories.forEach {
+        if (pathString.startsWith(it))
+          return true
+      }
+      return false
+    }
+
+    fun logProhibitedFilenameFailure(
+      errorToShow: String,
+      filePath: String
+    ) {
+      println("Prohibited filename pattern: [ROOT]/$filePath")
+      println("Failure message: $errorToShow")
+      println()
```

Comment:
rather than using extra println, we can use some escape pattern.

Reply:
we can use one println only.

Reply:
Done.